### PR TITLE
Added docker-compose as a workflow

### PR DESCRIPTION
**Pull request by tristan-swu**
I started to get 'play' going on my machine, but ran into some common issue (psycopq wouldn't install). One successful solution I've used in the past is to run python / django via docker-compose. This has some nice advantages:
 - installing dependencies (like psycopg) is much smoother
 - you can develop using the database you will use in production (in this case, Postgres 10.4)
 - volume mapping means the server will auto-reload on file changes.

I've kept the original documentation for setting up the project, but put it in a secondary doc. README.md has been updated with the new install workflow.

I removed `dotenv` as a dependency. This is now managed by default in docker-compose.yml. The readme has been updated to `source .env` it for non docker-compose workflows.

An example `.env` file is provided.

### CODE REVIEW

File: play/settings/base.py [hunk 1]
```diff
@@ -25,34 +23,24 @@ def get_env(key, default=None, allow_default=True):
...
-
-
 # Build paths inside the project like this: os.path.join(BASE_DIR, ...)
 BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
 
 # Get environment variable ENV from the system
 # default to production if it doesn't exist
 ENV = get_env('ENV', 'production', True)
 
-def is_local_env():
```
Comment (tristan-swu) [inline play/settings/base.py]: This function was not being used.

We _may_ want to consider using [DJANGO_SETTINGS_MODULE](https://docs.djangoproject.com/en/2.1/topics/settings/#designating-the-settings) in the future. I think its a nice way to provide variations for different environments.

File: Dockerfile [hunk 2]
```diff
@@ -29,4 +29,5 @@ COPY ./play \
     /app/
 
 EXPOSE 8000
-ENTRYPOINT [ "/app/entrypoint.sh" ]
```
Comment (tristan-swu) [inline Dockerfile]: `ENTRYPOINT` doesn't allow you to override using a `command` in docker-compose.yml, so this enables overriding for dev workflows.

File: play/settings/base.py [hunk 3]
```diff
@@ -125,26 +113,15 @@ def is_production_env():
 
 DATABASES = {
     'default': {
-        'ENGINE': 'django.db.backends.sqlite3',
```
Comment (tristan-swu) [inline play/settings/base.py]: SQLite is a nice project, but you can't beat testing on the same database you'll be using in production.

**Comment by dlsteuer** [comment]
Just a quick comment, we made a decision early on in the project to not do docker first workflow. I don't mind it as an alternative though if you want to adjust this accordingly

**Comment by tristan-swu** [comment]
I can restore the original README.md and link to the docker-compose workflow. 
I think there's a good case against using different dbs in dev vs production. What if I add a line in there about starting a postgres, so continuing to drop sqllite? Any objects there?

File: Dockerfile (hunk 2 above)
Comment (dlsteuer) [reply Dockerfile]: you should be able to with `entrypoint:` according to the docs https://docs.docker.com/compose/compose-file/. My concern here is the gke deployment will stop working with this change

**Comment by dlsteuer** [comment]
I understand that, the nicety with sqlite is you don't have to worry about anything other then a python environment, so no external database to setup, i don't mind having an override for postgres, and possibly even have travis use that, but for just running locally, sqlite, makes it dead simple

File: README.md [hunk 4]
```diff
@@ -10,6 +10,7 @@ This is the module that users of all types use to interact with the [game engine
 
 - On a mac? If you haven't already, install [brew](https://brew.sh/)
 - A [Github OAuth App](#github-oauth-app-configuration)
+- Postgres 10.4
```
Comment (dlsteuer) [inline README.md]: I don't think this is required unless you want to

Comment (tristan-swu) [reply README.md]: There's no mention of the databases being used in the readme, and maybe thats a future improvement. Removed for now.

**Comment by dlsteuer** [comment]
thanks @tristan-swu

**Pull request merged**